src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ExpressionHelper.cs

```diff
@@ -108,17 +99,24 @@ public static string GetExpressionText(LambdaExpression expression, ExpressionTe
             }
 
             // If parts start with "model", then strip that part away.
-            if (nameParts.Count > 0 && string.Equals(nameParts.Peek(), ".model", StringComparison.OrdinalIgnoreCase))
+            if (part == null || part.NodeType != ExpressionType.Parameter)
             {
-                nameParts.Pop();
+                var text = builder.ToString();
```

Comment:
@mnltejaswini I think you said you had an idea about avoiding this extra `string` allocation. Whether I'm remembering things correctly or not, let's get this change in. Deal with this later (if it can be dealt with).